config/routes.rb

```diff
@@ -4,6 +4,16 @@
   devise_scope :user do
     get '/users/sign_out' => 'devise/sessions#destroy'
   end
+  
+  namespace :api, default: {format: :json} do 
+    namespace :v1,  default: {format: :json} do
+      resources :users, only: [:index, :show] do
+      resources :posts, only: [:index, :show, :create] do
+        resources :comments, only: [:index, :create]
+      end
+    end
+  end
+end
```

Comment:
- [ ] <b>REQUIRED:</b> There are few complications in your routes. if you decided to use the controllers in this directory `/app/controllers/api/v1/`, then, there shouldn't be their duplications outside. 
![scrnli_3_19_2022_12-57-30 PM](https://user-images.githubusercontent.com/45217364/159118364-346975ce-586c-4c58-bb68-f9b39cbcc620.png)
and this is how the code should look like: 

```rb
  namespace :api, default: {format: :json} do 
    namespace :v1,  default: {format: :json} do
       post 'login', to: 'authentication#authenticate'
       post 'register', to: 'users#create'
       resources :users do
         resources :posts, only: [:index, :show, :create] do
           resources :comments, only: [:index, :create]
           resources :likes, only: [:create]
         end
       end
     end
  end
```

And if you decide to use those controllers on this directory `/app/controllers`, then the code should be something like this: 

```rb
 resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show, :new, :create] do
      resources :comments, only: [:new, :create]
      resources :likes, only: [:create]
    end
  end
  post 'login', to: 'authentication#authenticate'
  post 'register', to: 'users#create'
```